### PR TITLE
New import pipeline

### LINKED ISSUE #331: Rework OSM import pipeline
I don't like the current OSM import pipeline. It's complicated for portal maintainers and requires way more resources than just running the instance. I'm usually running osm2pgsql on my local machine with lots of CPU+RAM and pushing the data through an SSH tunnel into the postgres on the server. That's not ideal.

I also would love to be able to add and remove and update OSM regions in the platform itself. Have a list of available and a list of selected regions, and let a cronjob or manual task or whatever handle the importing.

So I thought we could rework the whole pipeline and semi-centralize the OSM data processing. We could convert the original OSM files to a OBS-specific file on a per-region basis (e.g. by Bundesland or Country, as geofabrik does it) and do this once on a high performance machine (such as my private PC), offline. Then host these files centralized (e.g. at `mapdata.openbikesensor.org`) for all platform maintainers to access. The application could then list available files (i.e. regions) in the frontend for admins, and have a routine to import one of those, directly from a HTTP stream into memory into the database. No high peak memory, no node/way geometry matching, no tempfiles or mmap, just a plain import into postgis.

The "only" thing that is not straightforward here is how to handle OSM way_id that appears in multiple regions. I expect however that we'll find that they will be identical, as it would make little sense for geofabrik to split the ways and assign the same ID on different pieces of data for their extracts. OSM splits huge ways anyway (e.g. the Danube, the longest river in europe, is IIRC split into more than 50 sub-ways). We'll need to investigate this and have a proper strategy for how to handle conflicting data. Probably the newest wins ;)

This would also make automatic updates much much easier...

Let's brainstorm.

### PR DESCRIPTION
**Pull request by opatut**
Fixes #331

### CODE REVIEW

**Comment by opatut** [comment]
@gluap Would you like to test this too? It works fine for me so far. I'd love to merge this into `next` too.

**Comment by gluap** [comment]
Adding to this reply while testing:
- Fixed the docker file that was still referencing the roads import lua.
- Preparing the msgpack worked.
- The iterative deletion of "only affected way IDs" on first import took ages for me until I eventually aborted it and instead ran ``DELETE FROM road;`` which took only seconds. No big surprise as I had germany-latest loaded in my test instance using osm2pgsql before.

**Review by gluap — COMMENTED** [review]
I tested this on my local deployment and it worked. Import was much faster than using osm2pgsql.

**Comment by gluap** [comment]
OK, so the new tables all come from the newer postgis that I am using.
In the process I also noticed that the migration ``create_table_road.py`` had a typo that we haven't noticed yet because on all previous installations the table was created by ``osm2pgsql``. I noticed because I had initialized my database from a backup (as a means of upgrading postges) where I did not have ``road`` dumped to save on space.

**Comment by gluap** [comment]
OK, last update before bed:
The road table we create with the alembic migration has the wrong indexes. Version from alembic:
```
                                       Table "public.road"
     Column     |       Type        | Collation | Nullable |               Default                
----------------+-------------------+-----------+----------+--------------------------------------
 way_id         | bigint            |           | not null | nextval('road_way_id_seq'::regclass)
 zone           | zone_type         |           |          | 
 name           | character varying |           |          | 
 geometry       | geometry          |           |          | 
 directionality | integer           |           |          | 
 oneway         | boolean           |           |          | 
 import_group   | character varying |           |          | 
 Indexes:
    "road_pkey" PRIMARY KEY, btree (way_id)
    "ix_road_geometry" btree (geometry)
    "ix_road_way_id" btree (way_id)
```

Version created by osm2psql:
```
obs=# \d road
                             Table "public.road"
     Column     |           Type            | Collation | Nullable | Default 
----------------+---------------------------+-----------+----------+---------
 way_id         | bigint                    |           | not null | 
 zone           | zone_type                 |           |          | 
 directionality | integer                   |           |          | 
 name           | text                      |           |          | 
 geometry       | geometry(LineString,3857) |           |          | 
 oneway         | boolean                   |           |          | 
Indexes:
    "road_geometry_idx" gist (geometry) WITH (fillfactor='100')
 
 ```
 
 Because of the different indexes the index creation on ``road_geometry`` fails when one has created the tables using alembic/upgrade.py:
 
 ```
 psycopg.errors.ProgramLimitExceeded: index row size 3064 exceeds btree version 4 maximum 2704 for index "ix_road_geometry"
DETAIL:  Index row references tuple (1466,1) in relation "road".
```

**Comment by gluap** [comment]
same thing for ``region``:
```
obs=# ALTER TABLE region ALTER COLUMN geometry TYPE geometry(GEOMETRY,3857);
ALTER TABLE
obs=# DROP INDEX ix_region_geometry; 
DROP INDEX
obs=# CREATE INDEX ix_region_geometry ON region USING GIST (geometry) WITH (FILLFACTOR=100);
```

**Comment by gluap** [comment]
Note that also compared to osm2pgsql the type of text columns has changed to "character varying"

**Comment by gluap** [comment]
It might also be an issue for the import that the order of columns is different.

File: api/tools/import_osm.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        unpacker = msgpack.Unpacker(f)
+        try:
+            while True:
+                type_id = unpacker.unpack()
+                data = unpacker.unpack()
+
+                if type_id == b"\x01":
+                    yield Road(*data)
+
+                # elif type_id == b"\x02":
```
Comment (gluap) [inline api/tools/import_osm.py:59]: O_o thats why I don't get regions on import :)

**Comment by gluap** [comment]
With the new dump/import scripts I can still import Germany in 4 minutes (but now including regions)

**Pull request merged**